packages/hardhat/test/integration-tests/periphery/controller-helper.ts

```diff
@@ -0,0 +1,135 @@
+  let dai: MockErc20
+  let weth: WETH9
+  let positionManager: Contract
+  let uniswapFactory: Contract
+  let oracle: Oracle
+  let controller: Controller
+  let wSqueethPool: Contract
+  let wSqueeth: WPowerPerp
+  let ethDaiPool: Contract
+  let controllerHelper: ControllerHelper
+  let shortSqueeth: ShortPowerPerp
+
+
+  this.beforeAll("Deploy uniswap protocol & setup uniswap pool", async() => {
+    const accounts = await ethers.getSigners();
+    const [_owner, _depositor, _feeRecipient ] = accounts;
+    owner = _owner;
+    depositor = _depositor;
+    feeRecipient = _feeRecipient
+    provider = ethers.provider
+
+    const { dai: daiToken, weth: wethToken } = await deployWETHAndDai()
+
+    dai = daiToken
+    weth = wethToken
+
+    const uniDeployments = await deployUniswapV3(weth)
+    positionManager = uniDeployments.positionManager
+    uniswapFactory = uniDeployments.uniswapFactory
+
+    // this will not deploy a new pool, only reuse old onces
+    const squeethDeployments = await deploySqueethCoreContracts(
+      weth,
+      dai, 
+      positionManager, 
+      uniswapFactory,
+      scaledStartingSqueethPrice,
+      startingEthPrice
+    )
+    controller = squeethDeployments.controller
+    wSqueeth = squeethDeployments.wsqueeth
+    oracle = squeethDeployments.oracle
+    shortSqueeth = squeethDeployments.shortSqueeth
+    wSqueethPool = squeethDeployments.wsqueethEthPool
+    ethDaiPool = squeethDeployments.ethDaiPool
+
+    await controller.connect(owner).setFeeRecipient(feeRecipient.address);
+    await controller.connect(owner).setFeeRate(100)
+    
+    const ControllerHelperContract = await ethers.getContractFactory("ControllerHelper");
+    controllerHelper = (await ControllerHelperContract.deploy(controller.address, oracle.address, shortSqueeth.address, wSqueethPool.address, wSqueeth.address, weth.address, uniswapFactory.address)) as ControllerHelper;
+  })
+
+  this.beforeAll("Seed pool liquidity", async() => {
+    // add liquidity
+
+    await addWethDaiLiquidity(
+      startingEthPrice,
+      ethers.utils.parseUnits('100'), // eth amount
+      owner.address,
+      dai,
+      weth,
+      positionManager
+    )
+    await provider.send("evm_increaseTime", [600])
+    await provider.send("evm_mine", [])
+
+    await addSqueethLiquidity(
+      scaledStartingSqueethPrice, 
+      '1000000',
+      '2000000', 
+      owner.address, 
+      wSqueeth, 
+      weth, 
+      positionManager, 
+      controller
+    )
+    await provider.send("evm_increaseTime", [600])
+    await provider.send("evm_mine", [])
+  })
+
+  describe("Mint short with flash deposit", async () => {
+    it("flash mint", async () => {      
+      const vaultId = await shortSqueeth.nextId();
+      // await controller.connect(depositor).updateOperator(vaultId, controllerHelper.address)
+
+      const normFactor = await controller.normalizationFactor()
+      const mintWSqueethAmount = ethers.utils.parseUnits('2')
```

Comment:
think should make this larger as we will have to update tests to take into account 6.9 eth min collateral (might as well make it work now)